lib/src/animations/chart_animations.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:flutter/material.dart';
+
+/// Handles updating [tweenSeries] with new data and manages the [animation].
+///
+/// To start listening to animation updates and update data along with notifying painters on each update
+/// call [animateDataUpdates].
+///
+/// To update the tween when new data is available and animate to the new state call [updateDataSeries].
+mixin ChartAnimations<T> on ChangeNotifier {
```

Comment:
Let's follow the common naming convention of ending mixin class names with `Mixin`. This way it's earlier to know that you need to mixin this class and not extend it.